.github/workflows/iroha2-dev-pr.yml

```diff
@@ -92,94 +94,44 @@ jobs:
           path: ${{ env.DOCKER_COMPOSE_PATH }}
-      - uses: taiki-e/install-action@nextest
-      - uses: taiki-e/install-action@cargo-llvm-cov
-      - name: Run unit tests (no default features)
+      - name: Download the rest of WASM samples
+        uses: actions/download-artifact@v4
+        with:
+          name: wasm_samples
+          path: ${{ env.WASM_SAMPLES_TARGET_DIR }}
+      - name: Install irohad
+        run: which irohad || cargo install --path crates/irohad --locked
+      - name: Test with no default features
         run: >
           mold --run cargo llvm-cov nextest
-          --no-fail-fast
-          --workspace --lib
           --no-default-features
-          --branch
-          --no-report
-      - name: Run unit tests (all features)
+          --branch --no-report
+      - name: Test with all features
         run: >
           mold --run cargo llvm-cov nextest
-          --no-fail-fast
-          --workspace --lib
           --all-features
-          --branch
-          --no-report
+          --branch --no-report
       - name: Generate lcov report
         run: cargo llvm-cov report --lcov --output-path lcov.info
       - name: Upload lcov report
         uses: actions/upload-artifact@v4
         with:
           name: report-coverage
           path: lcov.info
-
-  # include: iroha/tests/integration/
-  # exclude: iroha/tests/integration/extra_functional
-  integration:
-    runs-on: [self-hosted, Linux, iroha2]
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-09-09
-    needs: build_wasm_samples
-    timeout-minutes: 30
-    steps:
-      - uses: actions/checkout@v4
-      - name: Download executor.wasm
-        uses: actions/download-artifact@v4
-        with:
-          name: executor.wasm
-          path: ${{ env.DOCKER_COMPOSE_PATH }}
-      - name: Download the rest of WASM samples
-        uses: actions/download-artifact@v4
-        with:
-          name: wasm_samples
-          path: ${{ env.WASM_SAMPLES_TARGET_DIR }}
-      - uses: taiki-e/install-action@nextest
-      - name: Run integration tests, with all features
-        run: >
-          mold --run cargo nextest run 
-          --all-features
-          --no-fail-fast 
-          --failure-output immediate-final 
-          -E 'package(iroha) and test(integration) and not test(extra_functional)'
-
-  # include: iroha/tests/integration/extra_functional
-  extra_functional:
-    runs-on: [self-hosted, Linux, iroha2]
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-09-09
-    needs: build_wasm_samples
-    timeout-minutes: 60
-    steps:
-      - uses: actions/checkout@v4
-      - name: Download executor.wasm
-        uses: actions/download-artifact@v4
-        with:
-          name: executor.wasm
-          path: ${{ env.DOCKER_COMPOSE_PATH }}
-      - name: Download the rest of WASM samples
-        uses: actions/download-artifact@v4
+      - name: Upload test network artifacts
+        if: failure()
```

Comment:
This will run if any step fails for any reason, I think. I wonder if it's possible to run this only when `cargo nextest` returns a specific code for test failure

Reply:
Am I right that test network artifacts should be uploaded when `Test with no default features` and/or `Test with all features` steps are failed?

Reply:
Yes, do you see any problems with that?

Reply:
I think we can [add conditions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#failure-with-conditions) to upload only when specific steps failed

Reply:
Might be
```
      - name: Test with no default features
        id: no_feat_tests
        run: >
          mold --run cargo llvm-cov nextest
          --no-default-features
          --branch --no-report

      - name: Test with all features
        id: all_feat_tests
        run: >
          mold --run cargo llvm-cov nextest
          --all-features
          --branch --no-report
      - name: Upload test network artifacts
        if: ${{ (failure() && steps.no_feat_tests.outcome == 'failure') || (failure() && steps.all_feat_tests.outcome == 'failure') }}
        uses: actions/upload-artifact@v4
        with:
          name: test_network_runs
          path: ${{ env.TEST_NETWORK_TMP_DIR }}/irohad_test_network_*
          retention-days: 3
```

Reply:
Updated the policy accordingly